src/js/plugins/menu_toggle/plugin.menu_toggle.js

```diff
@@ -7,84 +8,92 @@
  * This is fired when user clicks on the fullscreen button on the menu
  */
 
-(function addFullScreenMenuToggler () {
+(function addMenuToggler () {
     jQuery.extend(BookReader.defaultOptions, {
-      enableFullScreenMenuToggle: true
+      enableMenuToggle: true
     });
   
-    function onEnterFullscreen (br) {
+    function setupNavForToggle (br) {
       var $menuTab = br.refs.$BRnav.children('.js-tooltip');
       $menuTab.css('display', 'none');
-  
-      var timeoutHandler = function onEnterTimeoutHandler () {
-        br.hideNavigation();
-        var menuManager = br.fullscreenMenu;
-  
-        registerEventHandlers(br);
-        menuManager.menuIsShowing = false;
-      }
-      setTimeout(timeoutHandler, 500);
+      registerEventHandlers(br);
     }
   
-    var onExitFullscreen = function onExitFullscreen (br) {
+    var removeToggleFromNav = function removeToggleFromNav (br) {
       var $menuTab = br.refs.$BRnav.children('.js-tooltip');
       $menuTab.css('display', 'block');
       removeEventHandlers(br);
     }
   
-    var removeEventHandlers = function removeEventHandlers(br) {
+    var removeEventHandlers = function removeEventHandlers (br) {
       if (br.refs.$brPageViewEl) {
         br.refs.$brPageViewEl[0].removeEventListener('click', toggleMenuIfCenterClick, true);
       }
       if (br.refs.$brTwoPageView) {
         br.refs.$brTwoPageView[0].removeEventListener('click', toggleMenuIfCenterClick, true);
       }
     }
+
+    var toggleNav = function toggleNav (br) {
+      var menuIsShowing = br.navigationIsVisible();
+      if (menuIsShowing) {
+        br.hideNavigation();
+      } else {
+        br.showNavigation();
+      }
+    }
   
     var toggleMenuIfCenterClick = function toggleMenuIfCenterClick(br, e) {
-      var menuManager = br.fullscreenMenu;
-      var bookWidth = e.currentTarget.offsetWidth;
-      var leftOffset = e.currentTarget.offsetLeft
-      var bookEndPageFlipArea = Math.round(bookWidth / 5);
+      var book = e.currentTarget;
+      if (!book) {
+        return;
+      }
```

Comment:
question: I don't think it's possible for `currentTarget` to be null?

Reply:
true. would rather be safe at this point since we are doing calculations against the target immediately after this line.